src/app/reexportr-src/reexportr.directive.ts

```diff
@@ -1,17 +1,39 @@
-import { Directive, Input, TemplateRef, ViewContainerRef } from "@angular/core";
+import {
+  Directive,
+  Input,
+  TemplateRef,
+  ViewContainerRef,
+  OnInit,
+  OnDestroy,
+  EmbeddedViewRef
+} from "@angular/core";
 
 @Directive({
   selector: "[reExportr]"
 })
-export class ReExportrDirective {
+export class ReExportrDirective implements OnInit, OnDestroy {
+  context = {
+    $implicit: undefined,
+    reExportr: undefined
+  };
+  embeddedView: EmbeddedViewRef<any>;
+
   constructor(private _tr: TemplateRef<any>, private _vcr: ViewContainerRef) {}
 
+  ngOnInit() {
+    this.embeddedView = this._vcr.createEmbeddedView(this._tr, this.context);
+  }
+
+  ngOnDestroy() {
+    this._vcr.clear();
+  }
+
   @Input()
   set reExportr(val: any) {
-    this._vcr.clear();
-    this._vcr.createEmbeddedView(this._tr, {
-      $implicit: val,
-      reExportr: val
-    });
+    this.context.reExportr = val;
+    this.context.$implicit = val;
+    if (this.embeddedView) {
+      this.embeddedView.detectChanges();
+    }
```

Comment:
See my previous comment and the stackblitz attached (inside the `ng-let.directive`), I don't think we need to manually call detectChanges? 

My assumption is that Angular takes over destroying, calling CD etc. on these embedded views we create?

Reply:
Hmm, you seem to be right! This is suprising to me: I expected that because I was keeping a persistent `context` object and mutating it that it wouldn't run CD properly. Interesting lesson to have learned from! I'll go ahead and update this with both changes